plrust-tests/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+    "#,
+        name = "create_contributors_pets",
+    );
+}
+
+#[cfg(any(test, feature = "pg_test"))]
+pub mod pg_test {
+    use once_cell::sync::Lazy;
+    use tempdir::TempDir;
+
+    static WORK_DIR: Lazy<String> = Lazy::new(|| {
+        let work_dir = TempDir::new("plrust-tests").expect("Couldn't create tempdir");
+        format!("plrust.work_dir='{}'", work_dir.path().display())
+    });
+    static LOG_LEVEL: &str = "plrust.tracing_level=trace";
+
+    static PLRUST_ALLOWED_DEPENDENCIES_FILE_NAME: &str = "allowed_deps.toml";
+    static PLRUST_ALLOWED_DEPENDENCIES_FILE_DIRECTORY: Lazy<TempDir> = Lazy::new(|| {
+        use std::io::Write;
+        let temp_allowed_deps_dir =
+            TempDir::new("plrust-allowed-deps").expect("Couldnt create tempdir");
+
+        let file_path = temp_allowed_deps_dir
+            .path()
+            .join(PLRUST_ALLOWED_DEPENDENCIES_FILE_NAME);
+        let mut allowed_deps = std::fs::File::create(&file_path).unwrap();
+        allowed_deps
+            .write_all(
+                r#"
+owo-colors = "=3.5.0"
+tokio = { version = "=1.19.2", features = ["rt", "net"]}
+plutonium = "*"
+"#
+                .as_bytes(),
+            )
+            .unwrap();
+
+        temp_allowed_deps_dir
+    });
+
+    static PLRUST_ALLOWED_DEPENDENCIES: Lazy<String> = Lazy::new(|| {
+        format!(
+            "plrust.allowed_dependencies='{}'",
+            PLRUST_ALLOWED_DEPENDENCIES_FILE_DIRECTORY
+                .path()
+                .join(PLRUST_ALLOWED_DEPENDENCIES_FILE_NAME)
+                .to_str()
+                .unwrap()
+        )
+    });
+
+    pub fn setup(_options: Vec<&str>) {
+        // perform one-off initialization when the pg_test framework starts
+    }
+
+    pub fn postgresql_conf_options() -> Vec<&'static str> {
+        vec![
+            &*WORK_DIR,
+            &*LOG_LEVEL,
+            &*PLRUST_ALLOWED_DEPENDENCIES,
+            "shared_preload_libraries='plrust'",
+        ]
+    }
+}
+
+
+/*
+#[cfg(any(test, feature = "pg_test"))]
+#[pg_schema]
+mod tests {
+    use pgrx::prelude::*;
+
+    #[pg_test]
+    fn test_plrust_works() -> spi::Result<()> {
+        Spi::run("CREATE FUNCTION test_plrust_works() RETURNS int LANGUAGE plrust AS $$ Ok(Some(1)) $$;")?;
+        let result = Spi::get_one::<i32>("SELECT test_plrust_works();");
+        assert_eq!(result, Ok(Some(1)));
+
+        Ok(()) 
+    }
+
+}
+
+/// This module is required by `cargo pgrx test` invocations.
+/// It must be visible at the root of your extension crate.
+#[cfg(test)]
+pub mod pg_test {
+    pub fn setup(_options: Vec<&str>) {
+        // perform one-off initialization when the pg_test framework starts
+    }
+
+    pub fn postgresql_conf_options() -> Vec<&'static str> {
+        // return any postgresql.conf settings that are required for your tests
+        vec![
+            "shared_preload_libraries = 'plrust'",
+            "plrust.work_dir = '/tmp/plrust-work-dir'"
+        ]
+    }
+}
+*/
```

Comment:
Pardon, but why is this commented out here?

Reply:
I believe it was a hold over from moving all the test here 
will be removed